src/ZODB/Connection.py

```diff
@@ -757,7 +758,11 @@ def afterCompletion(self, transaction):
         #    finalizing previous ones without calling begin.  We pass
         #    False to avoid possiblyt expensive sync calls to not
         #    penalize well-behaved applications that call begin.
-        self.newTransaction(transaction, False)
+        if hasattr(self._storage, 'afterCompletion'):
+            self._storage.afterCompletion()
```

Comment:
> but avoiding that seems like the responsibility of the storage. Ultimately, this is like an event that the storage can do with what it will.

I agree. There's not really anything else that can be done at this level.